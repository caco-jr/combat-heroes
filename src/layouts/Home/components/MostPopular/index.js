import React from 'react';

import { SectionTitle } from '../../../../components/Title/index.style';
import { CardList } from '../../../../components/Card/List/index.style';
import Card from '../../../../components/Card/Default';

const MostPopular = ({ data }) => {
  return (
    <section className="container">
      <SectionTitle>Most Popular</SectionTitle>

      <CardList>
        {data &&
          data.map((item, index) => {
            const { id, name, image } = item;

            return (
              <Card key={index} id={id} name={name} imageURL={image.url} />
            );
          })}
      </CardList>
    </section>
  );
};

export default MostPopular;
