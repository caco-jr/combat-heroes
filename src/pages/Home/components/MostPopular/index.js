import React from 'react';

import { CardList } from '../../../../components/Card/List/index.style';
import Card from '../../../../components/Card/Default';

const MostPopular = ({ data }) => {
  return (
    <section className="container">
      <h2>Most Popular</h2>

      <CardList>
        {data.map((item, index) => {
          const { id, name, image } = item.data.character;

          return <Card key={index} id={id} name={name} imageURL={image.url} />;
        })}
      </CardList>
    </section>
  );
};

export default MostPopular;
