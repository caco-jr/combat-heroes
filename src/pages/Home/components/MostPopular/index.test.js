import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import MostPopular from './index';

describe('With Enzyme', () => {
  //   it('Should have ', () => {
  //     const app = shallow(<MostPopular data={{ data: handleData() }} />);
  //     expect(app.find('Card')).toEqual(9);
  //   });
});

describe('With Snapshot Testing', () => {
  it('Rendering MostPopular', () => {
    const component = renderer.create(<MostPopular />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
