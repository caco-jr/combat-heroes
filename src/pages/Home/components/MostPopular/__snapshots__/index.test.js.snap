// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Rendering MostPopular 1`] = `
<section
  className="container"
>
  <h2>
    Most Popular
  </h2>
  <section
    className="indexstyle__CardList-sc-5v81d9-0 eIwYVw"
  />
</section>
`;
